public/app/plugins/panel/timeseries/TimeSeriesTooltip.tsx

```diff
@@ -61,7 +64,8 @@ export const TimeSeriesTooltip = ({
     seriesIdx,
     mode,
     sortOrder,
-    (field) => field.type === FieldType.number || field.type === FieldType.enum
+    (field) => field.type === FieldType.number || field.type === FieldType.enum,
+    (value) => (hideZeros ? value !== '0' : true)
```

Comment:
I can see where this is not ideal because it evaluates for every value. I went with this way to match the field filter above and because I can see this being a fairly likely place for the logic to get more complicated and account for NaNs, Infinity-s, etc.